File: app/src/main/java/hu/bme/szn/randomjokes/model/Joke.java
package hu.bme.szn.randomjokes.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Joke {

    @SerializedName("id")
    @Expose(serialize = false)
    private Long id;

    @SerializedName("type")
    @Expose
    private String type;

    @SerializedName("setup")
    @Expose
    private String setup;

    @SerializedName("body")
    @Expose
    private String punchline;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getSetup() {
        return setup;
    }

    public void setSetup(String setup) {
        this.setup = setup;
    }

    public String getPunchline() {
        return punchline;
    }

    public void setPunchline(String punchline) {
        this.punchline = punchline;
    }

}

File: app/src/main/java/hu/bme/szn/randomjokes/model/Joke.kt
package hu.bme.szn.randomjokes.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.SerializedName;

@Entity
class Joke(

    @SerializedName("setup")
    var setup: String,

    @SerializedName("punchline")
    var punchline: String,

    @SerializedName("type")
    var type: String,

    @SerializedName("id")
    @PrimaryKey
    val id: Long

)
